react-web/src/hooks/useComponentVisible.ts

```diff
@@ -0,0 +1,28 @@
+import { useState, useEffect, useRef } from 'react';
+
+export function useComponentVisible(initialIsVisible: boolean) {
+  const [isComponentVisible, setIsComponentVisible] = useState(initialIsVisible);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const ref = useRef<any>(null);
+
+  const handleClickOutside = (event: React.MouseEvent<HTMLElement>) => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    if (ref.current && !ref.current.contains(event.target)) {
+      setIsComponentVisible(false);
+    }
+  };
+
+  useEffect(() => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    document.addEventListener('click', handleClickOutside, true);
+    return () => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      document.removeEventListener('click', handleClickOutside, true);
+    };
+  }, []);
+
+  return { ref, isComponentVisible, setIsComponentVisible };
+}
```

Comment:
Instead of HTMLElement used <any | null> to be compatible with all types of elements

This scenario has not been handled, as it is not clear
> The useComponentVisible hook is a good way to handle the visibility of a component based on clicks outside of it. However, there's a potential issue with memory leaks when the component unmounts while an async operation is still in progress. To prevent this, you should cancel all subscriptions and asynchronous tasks in the cleanup function returned by useEffect.